testplan/web_ui/testing/src/Report/BatchReport/state/reportSelectors.js

```diff
@@ -0,0 +1,55 @@
+import { createSelector } from '@reduxjs/toolkit/dist/redux-toolkit.esm';
+import * as appSelectors from '../../../state/appSelectors';
+
+export const mkGetReportUid = () => st => st.uid;
+export const getReportUid = mkGetReportUid();
```

Comment:
```
export const getReportUid = st => st.uid;
export const mkGetReportUid = () => getReportUid;
```

Reply:
?

So this `mkGet...` and `get...` pattern is because each React component needs its own copy of a selector, for reasons explained [here](https://github.com/reduxjs/reselect#sharing-selectors-with-props-across-multiple-component-instances). The `mkGet..` selector creators return a new selector that a component can use independent of other components.

Non-React-component consumers don't need their own copy, so they'll all just share the same `get...` selector.

Also FYI this file is now [here](https://github.com/Morgan-Stanley/testplan/pull/451/files#diff-7f203f3b143aa55dc4eb73a6f5af98b7).